test/Datadog.Trace.ClrProfiler.IntegrationTests/Helpers/IisFixture.cs

```diff
@@ -1,30 +1,31 @@
 using System;
 using System.Diagnostics;
+using System.IO;
 using Datadog.Core.Tools;
 using Datadog.Trace.TestHelpers;
 
 namespace Datadog.Trace.ClrProfiler.IntegrationTests
 {
     public sealed class IisFixture : IDisposable
     {
-        private Process _iisExpress;
+        private (Process Process, string ConfigFile) _iisExpress;
 
         public MockTracerAgent Agent { get; private set; }
 
         public int HttpPort { get; private set; }
 
-        public void TryStartIis(TestHelper helper)
+        public void TryStartIis(TestHelper helper, bool classicMode)
         {
             lock (this)
             {
-                if (_iisExpress == null)
+                if (_iisExpress.Process == null)
```

Comment:
Does this allow the two sets of tests to run at the same time? I'd expect only one invocation of this to happen so only either an app pool in Integrated Mode is started or an app pool in Classic Mode is started

Reply:
The configuration file is copied to a temporary name, so it shouldn't cause any issue with multiple concurrent executions (at least, no more issues than it did before)

Reply:
I'm just surprised that we can execute this multiple times. It looks like if I call `TryStartIis(helper, false)` followed by `TryStartIis(helper, true)` then we'd only start IIS Express once in Integrated Mode so we wouldn't be testing Classic Mode

Reply:
Yeah but we don't 😛 You have for instance `public class AspNetWebApi2WithFeatureFlagCallTargetIntegrated` and `public class AspNetWebApi2WithFeatureFlagCallTargetClassic`. They're separate classes, so they don't share the instance of `IisFixture`

Reply:
The aspnet tests use an `IClassFixture<>`. That means the `IISFixture` is only reused for [within-class tests (i.e. theories)](https://xunit.net/docs/shared-context#class-fixture). As we change mode in different classes, this should work as far as I can tell.

That said, if possible (I'm not sure it is), it might be nice to add a check in the web application itself to assert it's running in the correct mode?

Reply:
Ah thanks, my understanding of the class fixtures wasn't accurate. Thanks for clarifying!